java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/JavaCompactionRunner.java

```diff
@@ -137,15 +139,16 @@ private List<CloseableIterator<Row>> createInputIterators(CompactionJob compacti
 
     public static CloseableIterator<Row> getMergingIterator(
             ObjectFactory objectFactory, Schema schema, CompactionJob compactionJob,
-            List<CloseableIterator<Row>> inputIterators) throws IteratorCreationException {
+            List<CloseableIterator<Row>> inputIterators, String filtersConfig) throws IteratorCreationException {
         CloseableIterator<Row> mergingIterator = new MergingIterator(schema, inputIterators);
 
         // Apply an iterator if one is provided
-        if (null != compactionJob.getIteratorClassName()) {
+        if (null != compactionJob.getIteratorClassName() || null != filtersConfig) {
```

Comment:
Applying this here seems inconsistent with the way iterators are currently applied to compaction jobs. The table properties are read in CompactionJobFactory, and held in fields on the job. This way the behaviour of a job is fixed at the time it is created.